### DISCUSSION TITLE
Getting commit-analyzer to play nicely with Github Squash & Merge

### DISCUSSION THREAD

**Discussion by good-idea** [semantic-release/semantic-release#2275] [Configuration]
Hi all!

`commit-analyzer` handles default squash commit messages (https://github.com/semantic-release/commit-analyzer/pull/66)

However, when I choose the Squash & Merge option in Github, it doesn't format it in the same way:

![image](https://user-images.githubusercontent.com/11514928/143963466-47f6d442-2e6f-44e0-b66d-e8263d20a85c.png)

Does anyone have an example repo with a semantic-release setup that works with Squashed & Merged PRs?

**Comment by gr2m** [comment]
I use `squash & merge` exclusively on all my repositories. Unfortunately, GitHub only takes the pull request title as squash message if there is more than one commit, at least for these cases you can use https://github.com/apps/semantic-pull-requests. 

If you want you can add https://github.com/apps/squash-commit-app, but that doesn't work on PRs coming from forks.

Either way, you have to be careful to use the correct commit message.

**Comment by good-idea** [comment]
Thanks for the quick reply, @gr2m . I'll probably end up using the semantic-pull-requests app.

My other idea was to
 - create an 'in-between' branch, such as `merge-main`
 - allow only merge commits from PRs
 - write an action that would, when a PR is merged to `merge-main`, it performs a squash using git (so as to get the standard squash message), then merges that into `main`
 
 However, i think my contributors would be more likely to make sure they get the scope & commit type correct on a PR title, maybe not so much when they are committing on their branch. In which case the app you linked to would be the best solution.

**Reply by gr2m** [reply]
The main benefit of `squash & merge` is that your contributors don't need to worry at all about commit message conventions. The PRs can be as messy as they need to be, all you need to worry about is to use the correct message when merging.

Your workaround could work, but sounds like overkill. I'd say give it a try without.
